argilla-server/src/argilla_server/api/handlers/v1/oauth2.py

```diff
@@ -31,38 +29,44 @@
 router = APIRouter(prefix="/oauth2", tags=["Authentication"])
 
 
-_USER_ROLE_ON_CREATION = UserRole.annotator
+def check_oauth_enabled_or_raise() -> None:
+    if not settings.oauth.enabled:
+        raise NotFoundError(message="OAuth2 is not enabled")
+
+
+def get_provider_by_name_or_raise(provider: str = Path()) -> OAuth2ClientProvider:
+    check_oauth_enabled_or_raise()
```

Comment:
```suggestion
    if not settings.oauth.enabled:
        raise NotFoundError(message="OAuth2 is not enabled")
```

So you can remove `check_oauth_enabled_or_raise` function.